core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -419,6 +419,94 @@ public static int getFirstIndexUsingBinarySearch(FixedLengthDimensionDataChunk d
     return -(low + 1);
   }
 
+  public static int[] getRangeIndexUsingBinarySearch(
```

Comment:
you are right, I  really done binary search even for getting the ranges previously, but yesterday I done performance test and found the  performance is not better than current logic. the binary search range has advantage only under the condition of  data array size is very long  and the repeated data is too much. But usually the data array size is 12000 for a chunk, not too long. So  the binary search range has no advantage and I decide to keep the current logic